core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -161,24 +171,25 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     // all future commits should be simple commits
     this.updateType = UpdateType.SIMPLE;
 
-    updateCurrentMetadata(response);
+    this.current = response.tableMetadata();
```

Comment:
Sorry if it wasn't clear before, but this should not blindly replace table metadata if it has not changed.